api/v1alpha1/common_test.go

```diff
@@ -9,6 +9,245 @@ import (
+			&configv1alpha1.TerraformRepository{
+				Spec: configv1alpha1.TerraformRepositorySpec{
+					OpenTofuConfig: configv1alpha1.OpenTofuConfig{
+						Enabled: &[]bool{true}[0],
+					},
+				},
+			},
+			&configv1alpha1.TerraformLayer{},
+			true,
+		},
+		{"OnlyLayerEnabling",
+			&configv1alpha1.TerraformRepository{},
+			&configv1alpha1.TerraformLayer{
+				Spec: configv1alpha1.TerraformLayerSpec{
+					OpenTofuConfig: configv1alpha1.OpenTofuConfig{
+						Enabled: &[]bool{true}[0],
+					},
+				},
+			},
+			true,
+		},
+		// Edge case: Merging config from repository and layer should
+		// never set several base execution tools to true at the same time.
+		{"OverrideRepositoryTerraformWithLayerOpenTofu",
+			&configv1alpha1.TerraformRepository{
+				Spec: configv1alpha1.TerraformRepositorySpec{
+					TerraformConfig: configv1alpha1.TerraformConfig{
+						Enabled: &[]bool{true}[0],
+					},
+				},
+			},
+			&configv1alpha1.TerraformLayer{
+				Spec: configv1alpha1.TerraformLayerSpec{
+					OpenTofuConfig: configv1alpha1.OpenTofuConfig{
+						Enabled: &[]bool{true}[0],
+					},
+				},
+			},
+			true,
+		},
+		{"OverrideRepositoryOpentofuWithLayerTerraform",
+			&configv1alpha1.TerraformRepository{
+				Spec: configv1alpha1.TerraformRepositorySpec{
+					OpenTofuConfig: configv1alpha1.OpenTofuConfig{
+						Enabled: &[]bool{true}[0],
+					},
+				},
+			},
+			&configv1alpha1.TerraformLayer{
+				Spec: configv1alpha1.TerraformLayerSpec{
+					TerraformConfig: configv1alpha1.TerraformConfig{
+						Enabled: &[]bool{true}[0],
+					},
+				},
+			},
+			false,
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			result := configv1alpha1.GetOpenTofuEnabled(tc.repository, tc.layer)
+			if tc.expected != result {
+				t.Errorf("different base exec enabled status computed: expected %t go %t", tc.expected, result)
+			}
+		})
+	}
+}
+
+func TestGetOpenTofuVersion(t *testing.T) {
+	tt := []struct {
+		name            string
+		repository      *configv1alpha1.TerraformRepository
+		layer           *configv1alpha1.TerraformLayer
+		expectedVersion string
+	}{
+		{
+			"OnlyRepositoryVersion",
+			&configv1alpha1.TerraformRepository{
+				Spec: configv1alpha1.TerraformRepositorySpec{
+					OpenTofuConfig: configv1alpha1.OpenTofuConfig{
+						Version: "1.0.1",
+					},
+				},
+			},
+			&configv1alpha1.TerraformLayer{},
+			"1.0.1",
+		},
+		{
+			"OnlyLayerVersion",
+			&configv1alpha1.TerraformRepository{},
+			&configv1alpha1.TerraformLayer{
+				Spec: configv1alpha1.TerraformLayerSpec{
+					OpenTofuConfig: configv1alpha1.OpenTofuConfig{
+						Version: "1.0.1",
+					},
+				},
+			},
+			"1.0.1",
+		},
+		{
+			"OverrideRepositoryWithLayer",
+			&configv1alpha1.TerraformRepository{
+				Spec: configv1alpha1.TerraformRepositorySpec{
+					OpenTofuConfig: configv1alpha1.OpenTofuConfig{
+						Version: "1.0.1",
+					},
+				},
+			},
+			&configv1alpha1.TerraformLayer{},
+			"1.0.1",
+		},
```

Comment:
I think this is a duplicate of `OnlyRepositoryVersion`.
The "correct" `OverrideRepositoryWithLayer` is a few lines below.